### DISCUSSION TITLE
How to build a Malt Material Asset Library

### DISCUSSION THREAD

**Discussion by negicless** [bnpr/Malt#530] [Q&A]
Hey malt community,
I have a question for you guys.

If I want to make a library of custom malt materials, 
what is the best way to do so?

I made a lot of custom malt shaders/materials and I want to use them across different 
blender files via Asset Browser.

But everytime I add some asset, there will be errors in the malt node tree and it won't funtion as expected.

The custom shader contains a couple of sub node group.

Has anyone done it before?

Some tips will be very helpful!

Thanks

**Comment by lilschneezey** [comment]
I've had success. its just the normal process. The only gotcha has been that Malt must be turned on before you drop a material into a scene.
I have not had success with simple copy-and-paste of nodes. It seems to work the first time, but not the second. Malt materials seem to blow up.
I have not had success with making a malt material single-user. Malt materials seem to blow up. I've had to use the copy button to the right of the in-use counter.

**Reply by negicless** [reply]
Hey lilschneezey, 
thanks for replying.
I turn on malt every time I drop a material.
if there is already a malt material using a node group that has the same name as the dropped material,
it will mess up with the dropped material.
I am trying hard to rename every material and its malt node tree and node group tree ( really huge work) 
but after renaming the node group, the parameters disappear and I have to re-enter everything.'
do you have any solution to that?

**Reply by lilschneezey** [reply]
That is interesting... I can't test right now, just speculate. I wonder if it is something similar to what I experienced when clicking the single-user button.  Try using one of the different modes on the asset browser (center top drop down menu), like link, append, or append-reuse. I have not played with the three nodes to see what works best.

My workflow has been to drag-and-drop the material into my blend file (append I think?).  If I want to use the material again, I select it in the material tab of the properties window. no issues. I also link materials between objects, no issues.

If I want to use an existing malt material node tree as a template to create a new material, I choose the malt node I want to reuse in the material tab, then press the copy button (not the single-user button).
